### PR TITLE
kotlin: add conversion from RetryPolicy to headers

### PR DESCRIPTION
**Pull request by buildbreaker**
Add conversion from RetryPolicy to headers

Changes also made to make internal classes visible

Signed-off-by: Alan Chiu <achiu@lyft.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Add conversion from RetryPolicy to headers
Risk Level: low
Testing: unit
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: bazel/kotlin_test.bzl [hunk 1]
```diff
@@ -18,11 +18,17 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_test")
 # )
 #
 def envoy_mobile_kt_test(name, srcs, deps = []):
+    # This is to work around the issue where we have specific implementation functionality which
+    # we want to avoid comsumers to use but we want to unit test
+    dep_srcs = []
+    for dep in deps:
+        dep_srcs.append(dep + "_srcs")
```
Comment (rebello95) [inline bazel/kotlin_test.bzl]: Why is this necessary if the file in question is being included in the dependency?

Comment (buildbreaker) [reply bazel/kotlin_test.bzl]: In Kotlin, there's a concept of `internal` which allows for module level visibility.

The problem here is that in Bazel, we're using the library target as a dependency (so the test itself doesn't get visibility for `internal` entities). The reason this works is because we're basically leveraging the workaround in a [previous pr](https://github.com/lyft/envoy-mobile/pull/264) and use those defined `filegroup`s as source files in the test. Since it's now sources within the test target, we are able to access `internal` entities.

The other concept that I would like to call out is that we want to keep this entity (the mapper) an `internal` visible entity because we don't necessarily want consumers to use it. We also don't want to make it strictly private since that would make testing this behavior more complex (we'll have to do some integration black box testing or something along those lines).

File: library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt [hunk 2]
```diff
@@ -8,19 +8,19 @@ package io.envoyproxy.envoymobile
  * @param perRetryTimeoutMs Timeout (in milliseconds) to apply to each retry.
  */
 data class RetryPolicy(
-    val maxRetryCount: Int,
-    val retryOn: List<RetryRule>,
-    val perRetryTimeoutMs: Long?
+  val maxRetryCount: Int,
+  val retryOn: List<RetryRule>,
+  val perRetryTimeoutMs: Long? = null
```
Comment (buildbreaker) [inline library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt]: @rebello95 How do you feel about this default to null?

File: bazel/kotlin_test.bzl [hunk 3]
```diff
@@ -6,6 +6,7 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_test")
 # 1. Avoiding the need to declare the test_class which requires a fully qualified class name (example below)
 # 2. Avoiding the need to redeclare common unit testing dependencies like JUnit
 # 3. Ability to run more than one test file per target
+# 4. Ability to test internal entities which are envoy mobile dependencies
```
Comment (rebello95) [inline bazel/kotlin_test.bzl]: This doesn't quite make sense

File: bazel/kotlin_test.bzl [hunk 4]
```diff
@@ -18,11 +19,19 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_test")
 # )
 #
 def envoy_mobile_kt_test(name, srcs, deps = []):
+    # This is to work around the issue where we have specific implementation functionality which
+    # we want to avoid comsumers to use but we want to unit test
+    dep_srcs = []
+    for dep in deps:
+        # We'll resolve only the targets in `//library/kotlin/src/io/envoyproxy/envoymobile`
```
Comment (rebello95) [inline bazel/kotlin_test.bzl]: Would like @keith to check this file's changes

File: library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt (hunk 2 above)
Comment (rebello95) [reply library/kotlin/src/io/envoyproxy/envoymobile/RetryPolicy.kt]: Seems fine

File: bazel/kotlin_test.bzl (hunk 3 above)
Comment (buildbreaker) [reply bazel/kotlin_test.bzl]: `Ability to test internal envoy mobile entities`?

Comment (rebello95) [reply bazel/kotlin_test.bzl]: yea that would work
